### PR TITLE
Support multi-caret when inserting doc comments

### PR DESCRIPTION
**Pull request by Youssef1313**
gif

Fixes https://github.com/dotnet/roslyn/issues/54391

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
I'm ok with this (esp. with how minimal the code is).  but would def want a test for this :)

**Comment by CyrusNajmabadi** [comment]
Is thsi fixing some reported issue?

**Comment by Youssef1313** [comment]
> Is thsi fixing some reported issue?

Yeah I forgot to add it. Will link it from PR description: https://github.com/dotnet/roslyn/issues/54391

**Comment by Youssef1313** [comment]
Test infrastructure doesn't seem to be working with multi-caret. I'll look into how to fix that.

**Comment by Youssef1313** [comment]
@CyrusNajmabadi I was unable to write a test. Can you take a look so that we can get it merged? Thanks!

File: src/EditorFeatures/Core/Implementation/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs [hunk 1]
```diff
@@ -94,16 +95,23 @@ private static void ApplySnippet(DocumentationCommentSnippet snippet, ITextBuffe
...
+                    ApplySnippet(snippet, subjectBuffer, textView);
+                    returnValue = true;
+                }
             }
 
-            return false;
+            return returnValue;
         }
 
+
```
Comment (Youssef1313) [inline src/EditorFeatures/Core/Implementation/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs:114]:
```suggestion
```

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Can you take a look please?

**Comment by sharwell** [comment]
@Youssef1313 should be possible to write an integration test for this in the new integration test project

**Review by sharwell — CHANGES_REQUESTED** [review]
An integration test should be easy in the new integration test project

**Comment by Youssef1313** [comment]
@sharwell Which project? Can you point me please? Thanks!

**Comment by sharwell** [comment]
image

**Comment by sharwell** [comment]
The current tests have examples for how to do a bunch of different things. If you don't see a similar example for something you need, feel free to ask.

**Comment by Youssef1313** [comment]
@sharwell I couldn't find a test that does multi-caret. When I tried to do it I got `System.ArgumentException : Saw multiple occurrences of $$`

```csharp
        [IdeFact]
        [WorkItem(54391, "https://github.com/dotnet/roslyn/issues/54391")]
        public async Task TypingCharacter_MultiCaret()
        {
            var code =
@"
//$$
class C1 { }

//$$
class C2 { }

//$$
class C3 { }
";
            await SetUpEditorAsync(code, HangMitigatingCancellationToken);
            await TestServices.Input.SendAsync('/');
            var expected =
@"
/// <summary>
/// $$
/// </summary>
class C1 { }

/// <summary>
///
/// </summary>
class C2 { }

/// <summary>
///
/// </summary>
class C3 { }
";

            await TestServices.EditorVerifier.TextContainsAsync(expected, cancellationToken: HangMitigatingCancellationToken);
        }
```

**Comment by Youssef1313** [comment]
Also tried:

```csharp
//$$
class C1 { }

//[||]
class C2 { }

//[||]
class C3 { }
```

But fails with `System.Runtime.InteropServices.COMException : Error HRESULT E_FAIL has been returned from a call to a COM component.`

**Comment by davidwengier** [comment]
@Youssef1313 I took the liberty of updating the test. The issues were:

* The test framework didn't support multi-caret, so I added some basic support
* You didn't have a constructor in your test class, so didn't get a solution and project automatically created, so when calling `SetupEditorAsync` the COM exception occurred, because there wasn't an editor to setup.
* You weren't asserting the caret position, so the test was expecting to see `$$` in the actual editor
* You missed a couple of bits of whitespace

All in all though, you did a great job (unsurprisingly 😁)

**Comment by Youssef1313** [comment]
@davidwengier Thanks a lot!!

**Comment by MartyIX** [comment]
@Youssef1313 Do you have more up-to-date gif? Just curious. ;-)

**Comment by Youssef1313** [comment]
@MartyIX Oh looks like the software I used produced a bad GIF 😄 

Will build the branch and update it soon.

**Review by sharwell — APPROVED** [review]

**Pull request merged**